qiskit_superstaq/superstaq_backend.py

```diff
@@ -11,13 +11,57 @@
 # Any modifications or derivative works of this code must retain this
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
+import re
 from typing import Any, Dict, List, Optional, Union
 
 import qiskit
 
 import qiskit_superstaq as qss
 
 
+def validate_target(target: str) -> None:
+    vendor_prefixes = [
+        "aqt",
+        "aws",
+        "cq",
+        "d-wave",
+        "hqs",
+        "ibmq",
+        "ionq",
+        "neutral",
+        "rigetti",
+        "sandia",
+        "ss",
+    ]
+
+    target_device_types = ["qpu", "simulator"]
+
+    # Check valid format
+    pattern = re.compile("^([A-Za-z0-9-]+)_([A-Za-z0-9-.]+)_([a-z]+)")
+    if not pattern.fullmatch(target):
+        raise ValueError(
+            f"{target} does not have a valid string format. "
+            f"Valid backend strings should be in the form: "
+            f"provider_devicename_devicetype, e.g. ibmq_lagos_qpu."
+        )
+
+    prefix, _, device_type = target.split("_")
```

Comment:
fwiw you could just get vendor and device_type from the regex:
```suggestion
    match = re.fullmatch("([A-Za-z0-9-]+)_([A-Za-z0-9-.]+)_([a-z]+)", target)
    if not match:
        raise ValueError(
            f"{target} does not have a valid string format. "
            f"Valid backend strings should be in the form: "
            f"provider_devicename_devicetype, e.g. ibmq_lagos_qpu."
        )

    prefix, _, device_type = match.groups()
```